core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/signup/SignUpViewModel.kt

```diff
@@ -0,0 +1,429 @@
+
+    val genderOptions: List<MessageIdOption2> = PersonConstants.GENDER_MESSAGE_IDS_AND_UNSET,
+
+    val personPicture: PersonPicture? = null,
+
+    val registrationMode: Int = 0,
+
+    val usernameError: String? = null,
+
+    val fullName: String? = null,
+
+    val passwordError: String? = null,
+
+    val dateOfBirthError: String? = null,
+
+    val parentContactError: String? = null,
+
+    val genderError: String? = null,
+
+    val firstNameError: String? = null,
+
+    val lastNameError: String? = null,
+
+    val isParent: Boolean = false,
+
+    val isTeacher: Boolean = false,
+
+    val signupWithPasskey: Boolean = true,
+
+    val showCreatePasskeyPrompt: Boolean? = false,
+
+    val doorNodeId: String? = null,
+
+    val serverUrl_: String? = null,
+
+    val showOtherOption: Boolean = true
+) {
+
+
+}
+
+class SignUpViewModel(
+    di: DI,
+    savedStateHandle: UstadSavedStateHandle,
+    destName: String = DEST_NAME
+) : UstadEditViewModel(di, savedStateHandle, destName) {
+
+    private val _uiState: MutableStateFlow<SignUpUiState> = MutableStateFlow(SignUpUiState())
+
+    private val createPasskeyUseCase: CreatePasskeyUseCase? by instanceOrNull()
+
+    private var nextDestination: String =
+        savedStateHandle[UstadView.ARG_NEXT] ?: ClazzListViewModel.DEST_NAME_HOME
+
+    val uiState: Flow<SignUpUiState> = _uiState.asStateFlow()
+
+    private val apiUrlConfig: SystemUrlConfig by instance()
+
+    private val getLocalAccountsSupportedUseCase: GetLocalAccountsSupportedUseCase by instance()
+
+    private val serverUrl = apiUrlConfig.newPersonalAccountsLearningSpaceUrl ?: "http://localhost"
+
+    val addNewPersonUseCase: AddNewPersonUseCase = di.on(LearningSpace(serverUrl)).direct.instance()
+
+    private val genderConfig: GenderConfig by instance()
+
+    private val enqueueSavePictureUseCase: EnqueueSavePictureUseCase by
+    on(LearningSpace(serverUrl)).instance()
+
+
+    init {
+        loadingState = LoadingUiState.INDETERMINATE
+        val title =
+            systemImpl.getString(MR.strings.create_account)
+
+        _appUiState.update {
+            AppUiState(
+                title = title,
+                userAccountIconVisible = false,
+                hideBottomNavigation = true,
+            )
+        }
+        _uiState.update { prev ->
+            prev.copy(
+                genderOptions = genderConfig.genderMessageIdsAndUnset,
+                showCreatePasskeyPrompt = false,
+                person = Person(
+                    dateOfBirth = savedStateHandle[PersonEditViewModel.ARG_DATE_OF_BIRTH]?.toLong()
+                        ?: 0L
+                ),
+                serverUrl_ = serverUrl,
+                signupWithPasskey = createPasskeyUseCase != null,
+                showOtherOption = createPasskeyUseCase == null && getLocalAccountsSupportedUseCase.invoke()
+
+            )
+        }
+        if (savedStateHandle[SIGN_WITH_USERNAME_AND_PASSWORD] == "true") {
```

Comment:
Why would this need to be passed as an argument? It can be checked anytime by getting the passkey use case via instance or null, and then checking if that is not null

Reply:
i have done this because if passkey is supported and then user selected other option and in then selected signup with username and password then i am passing this argument .